File: wallet-kit/src/main/java/bitcoin/walllet/kit/struct/Inv.java
package bitcoin.walllet.kit.struct;

import java.io.IOException;

import bitcoin.walllet.kit.common.io.BitcoinInput;

/**
 * Inventory array.
 *
 * @author Michael Liao
 */
public class Inv {

    InvVect[] inventory;

    public Inv() {
    }

    public Inv(BitcoinInput input) throws IOException {
        long count = input.readVarInt(); // do not store count
        this.inventory = new InvVect[(int) count];
        for (int i = 0; i < this.inventory.length; i++) {
            this.inventory[i] = new InvVect(input);
        }
    }
}

File: wallet-kit/src/main/kotlin/bitcoin/wallet/kit/models/Inv.kt
package bitcoin.wallet.kit.models

import bitcoin.walllet.kit.io.BitcoinInput
import java.io.IOException

class Inv {

    var inventory: Array<InventoryItem>

    @Throws(IOException::class)
    constructor(input: BitcoinInput) {
        val count = input.readVarInt() // do not store count
        inventory = Array(count.toInt()) {
            InventoryItem(input)
        }
    }
}
